server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -365,24 +368,73 @@ public void merge(IndexMetadata indexMetadata, MergeReason reason) {
         }
     }
 
+    /**
+     * Merging the provided mappings. Actual merging is done in the raw, non-parsed, form of the mappings. This allows to do a proper bulk
+     * merge, where parsing is done only when all raw mapping settings are already merged.
+     */
+    public DocumentMapper merge(String type, List<CompressedXContent> mappingSources, MergeReason reason) {
+        final DocumentMapper currentMapper = this.mapper;
+        if (currentMapper != null && mappingSources.size() == 1 && currentMapper.mappingSource().equals(mappingSources.get(0))) {
+            return currentMapper;
+        }
+
+        Map<String, Object> mergedRawMapping = null;
+        for (CompressedXContent mappingSource : mappingSources) {
+            Map<String, Object> rawMapping = MappingParser.convertToMap(mappingSource);
+            if (mergedRawMapping == null) {
+                mergedRawMapping = rawMapping;
+            } else {
+                XContentHelper.merge(mergedRawMapping, rawMapping, ((parent, key, oldValue, newValue) -> {
+                    switch (key) {
+                        case "type" -> {
+                            // todo: verify this check is valid
+                            if (oldValue.equals(newValue) == false
+                                && (OBJECT_FIELD_TYPES.contains(String.valueOf(oldValue))
+                                    || OBJECT_FIELD_TYPES.contains(String.valueOf(newValue)))) {
+                                throw new MapperParsingException(
+                                    "can't merge a non object mapping [" + parent + "] with an object mapping"
+                                );
+                            }
+                        }
+                        case "subobjects" -> {
+                            if (oldValue.equals(newValue) == false) {
```

Comment:
Well, those are two contradicting  _**explicit**_ settings, so I figured we should assume that this is an error. Don't we prefer to let everyone this contradiction was introduced ASAP, instead of being surprised about the actual mappings?